Microsoft.AVS.VMFS/Microsoft.AVS.VMFS.psm1

```diff
@@ -1940,4 +1940,59 @@ function Clear-DisconnectedIscsiTargets {
             $VMHost | Get-VMHostStorage -RescanAllHba -RescanVmfs
         }
     }
+}
+
+<#
+    .SYNOPSIS
+     This function tests connection from a VMHost to remote target address.
+    
+    .PARAMETER VMHostName
+     VMHost name
+    
+    .PARAMETER TargetIpAddress
+     Target IP Address
+    
+    .PARAMETER TargetPort
+     Target Port
+    
+    .EXAMPLE
+        Test-ConnectionFromVMHost -VMHostName "esxi-01" -TargetIpAddress 10.0.0.1 -TargetPort 443"
+#>
+function Test-ConnectionFromVMHost {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false, AutomationOnly = $true)]
+    Param
+    (
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'VMHost name')]
+        [String] $VMHostName,
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Target IP Address')]
+        [String] $TargetIpAddress,
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Target Port')]
+        [int] $TargetPort   
```

Comment:
For security reasons, it would be better to only allow a specific list of ports for testing the connectivity. Other than port 3260, is there any others that would require a check?

Reply:
The main ports we need are 3260, 8084 (for VASA), and 8009 (for NVMe)
I could see that checking port 443 might become valuable in the future but we do not need that now.

Reply:
I added code to limit the target port that can be used